examples/tutorials/core_api/3_hpsearch.py

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python3
+
+"""
+Stage 3: Let's add hyperparameter search to our model.
+"""
+
+import logging
+import sys
+import time
+
+import determined as det
+
+def save_state(x, latest_batch, trial_id, path):
+    with path.joinpath("state").open("w") as f:
+        f.write(f"{x},{latest_batch},{trial_id}")
+
+def load_state(trial_id, path):
+    import pathlib
+    path = pathlib.Path(path)
+    with path.joinpath("state").open("r") as f:
+        x, latest_batch, ckpt_trial_id = [int(field) for field in f.read().split(",")]
+    if ckpt_trial_id == trial_id:
+        return x, latest_batch
+    else:
+        return x, 0
+
+def main(core_context, latest_checkpoint, trial_id, increment_by):
+    x = 0
+
+    starting_batch = 0
+    if latest_checkpoint is not None:
+        with core_context.checkpoint.restore_path(latest_checkpoint) as path:
+            x, starting_batch = load_state(trial_id, path)
+
+    # NEW: iterate through the core_context.searcher.operations() to decide how long to train for.
+    batch = starting_batch
+    last_checkpoint_batch = None
+    for op in core_context.searcher.operations():
+        # NEW: we will use a while loop for easier accounting of absolute lengths.
+        while batch < op.length:
+            x += increment_by
+            time.sleep(.1)
+            print("x is now", x)
+            batch += 1
+            if batch % 10 == 0:
```

Comment:
this used to be `== 9` in previous stages

Reply:
yeah, this was intentional.    The way things were counted changed slightly so this needed to as well.